### PR TITLE
be able to run defaulter-gen tests via `go test`

### PR DESCRIPTION
**Pull request by alexzielenski**
Continuation of #240 which went stale and closed.

> Files whose names begin with "_" are ignored by go test so none of the defaulter-gen tests were running in CI.
> [Some] tests failed because some packages missed a fake_deepcopy_conversion.go file. This is necessary because types registered in the scheme need to implement the DeepCopyObject() method.

Took into account some of the feedback provided on that PR, keeping `_output-tests` but adding its own go.mod so there can be a dependency on k/k packages directly without forming a cycle.

Changes from initial PR:
* Move `output_tests` back to `_output_tests`
* Added go.mod to _output_tests directory
* Rename package path from `_output_tests` to `output_tests` - otherwise retaining underscore in go package name caused ambiguous import error with gengo's file tree:
The module name itself still needed the underscore removed otherwise would get this error:
```console
ambiguous import: found package k8s.io/gengo/examples/defaulter-gen/_output_tests/empty in multiple modules:
        k8s.io/gengo v0.0.0 ($GOMOD/k8s.io/gengo@v0.0.0/examples/defaulter-gen/_output_tests/empty)
        k8s.io/gengo/examples/defaulter-gen/_output_tests (/Users/alex/go/src/k8s.io/gengo/examples/defaulter-gen/_output_tests/empty)
ambiguous import: found package k8s.io/gengo/examples/defaulter-gen/_output_tests/marker in multiple modules:
        k8s.io/gengo v0.0.0 ($GOMOD/k8s.io/gengo@v0.0.0/examples/defaulter-gen/_output_tests/marker)
        k8s.io/gengo/examples/defaulter-gen/_output_tests (/Users/alex/go/src/k8s.io/gengo/examples/defaulter-gen/_output_tests/marker)
ambiguous import: found package k8s.io/gengo/examples/defaulter-gen/_output_tests/marker/external in multiple modules:
        k8s.io/gengo v0.0.0 ($GOMOD/k8s.io/gengo@v0.0.0/examples/defaulter-gen/_output_tests/marker/external)
        k8s.io/gengo/examples/defaulter-gen/_output_tests (/Users/alex/go/src/k8s.io/gengo/examples/defaulter-gen/_output_tests/marker/external)
ambiguous import: found package k8s.io/gengo/examples/defaulter-gen/_output_tests/marker/external/external in multiple modules:
        k8s.io/gengo v0.0.0 ($GOMOD/k8s.io/gengo@v0.0.0/examples/defaulter-gen/_output_tests/marker/external/external)
        k8s.io/gengo/examples/defaulter-gen/_output_tests (/Users/alex/go/src/k8s.io/gengo/examples/defaulter-gen/_output_tests/marker/external/external)
ambiguous import: found package k8s.io/gengo/examples/defaulter-gen/_output_tests/pointer in multiple modules:
        k8s.io/gengo v0.0.0 ($GOMOD/k8s.io/gengo@v0.0.0/examples/defaulter-gen/_output_tests/pointer)
        k8s.io/gengo/examples/defaulter-gen/_output_tests (/Users/alex/go/src/k8s.io/gengo/examples/defaulter-gen/_output_tests/pointer)
ambiguous import: found package k8s.io/gengo/examples/defaulter-gen/_output_tests/slices in multiple modules:
        k8s.io/gengo v0.0.0 ($GOMOD/k8s.io/gengo@v0.0.0/examples/defaulter-gen/_output_tests/slices)
        k8s.io/gengo/examples/defaulter-gen/_output_tests (/Users/alex/go/src/k8s.io/gengo/examples/defaulter-gen/_output_tests/slices)
ambiguous import: found package k8s.io/gengo/examples/defaulter-gen/_output_tests/wholepkg in multiple modules:
        k8s.io/gengo v0.0.0 ($GOMOD/k8s.io/gengo@v0.0.0/examples/defaulter-gen/_output_tests/wholepkg)
        k8s.io/gengo/examples/defaulter-gen/_output_tests (/Users/alex/go/src/k8s.io/gengo/examples/defaulter-gen/_output_tests/wholepkg)
make: *** [test] Error 1
```

I think since this PR adds a new go module CI would need to be updated to run it. This PR makes `go test ./...` from within `_output_tests` correctly run without hacks.

### CODE REVIEW

File: examples/defaulter-gen/Makefile [hunk 1]
```diff
@@ -2,7 +2,7 @@ TOOL=defaulter-gen
 
 .PHONY : gen test verify git_verify_clean git_verify_no_change
 test:
-	go test ./_output_tests/marker/...
+	cd _output_tests; go test ./...
```
Comment (apelisse) [inline examples/defaulter-gen/Makefile:5]: nit but maybe `&&`?

**Review by apelisse — COMMENTED** [review]
Thanks, that's definitely confusing. Can we get a comment somewhere to indicate what's going on without having to go through the github history?

**Comment by alexzielenski** [comment]
updated PR description

File: examples/defaulter-gen/Makefile (hunk 1 above)
Comment (alexzielenski) [reply examples/defaulter-gen/Makefile:5]: Was following the same pattern as the commands for `make gen`. Should both change?

https://github.com/kubernetes/gengo/blob/1d7a6dd710b46db3d7a2f0665f92d7c460c90f74/examples/defaulter-gen/Makefile#L7-L10

Comment (apelisse) [reply examples/defaulter-gen/Makefile:5]: I would go ahead and change both yeah, thanks.

**Pull request merged**